package-res/reportviewer/reportviewer.js

```diff
@@ -863,30 +875,44 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
                 } else if (resultJson.status == "CONTENT_AVAILABLE") {
+
+                  var handleContAvailCallback = dojo.hitch(this, function(result2) {
+                    resultJson2 = JSON.parse(result2);
+                    if(resultJson2.status == "QUEUED" || resultJson2.status == "WORKING") {
+                      var urlStatus2 = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + resultJson2.uuid + '/status';
+                      setTimeout(function(){ pentahoGet(urlStatus2, "", handleContAvailCallback); }, this.reportPrompt._pollingInterval);
+                    } else if (resultJson2.status == "FINISHED") {
+                      var urlContent2 = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + resultJson2.uuid + '/content';
+                      logger && logger.log("Will set iframe url to " + urlContent2.substr(0, 50) + "... ");
+
+                      $('#hiddenReportContentForm').attr("action", urlContent2);
+                      $('#hiddenReportContentForm').submit();
+                      $('#reportContent').attr("data-src", urlContent2);
+                      this._updatedIFrameSrc = true;
+                      dlg.hide();
+                      isIframeContentSet = true;
+                    }
+                  });
+
                   if(isFirstContAvStatus) {
                     isFirstContAvStatus = false;
-                    var handleContAvailCallback = dojo.hitch(this, function(result2) {
-                      resultJson2 = JSON.parse(result2);
-                      if(resultJson2.status == "QUEUED" || resultJson2.status == "WORKING") {
-                        var urlStatus2 = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + resultJson2.uuid + '/status';
-                        setTimeout(function(){ pentahoGet(urlStatus2, "", handleContAvailCallback); }, this.reportPrompt._pollingInterval);
-                      } else if (resultJson2.status == "FINISHED") {
-                        var urlContent2 = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + resultJson2.uuid + '/content';
-                        logger && logger.log("Will set iframe url to " + urlContent2.substr(0, 50) + "... ");
- 
-                        $('#hiddenReportContentForm').attr("action", urlContent2);
-                        $('#hiddenReportContentForm').submit();
-                        $('#reportContent').attr("data-src", urlContent2);
-                        this._updatedIFrameSrc = true;
-                        dlg.hide();
-                        isIframeContentSet = true;
-                      }
-                    });
-                    //get first page
-                    pentahoGet('reportjob', url.substring(url.lastIndexOf("/report?")+"/report?".length, url.length), handleContAvailCallback, 'text/text');
+
+                    if(this._currentStoredPagesCount > me.view._getAcceptedPage()){
```

Comment:
Why do you use me.view._getAcceptedPage here and this.requestedPage later. Shouldnt that be consistent, for instance by ensuring that you always use "this._requestedPage"?